rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/AnalyzeJSPFileRuleProvider.java

```diff
@@ -0,0 +1,174 @@
+import org.jboss.windup.ast.java.data.ClassReference;
+import org.jboss.windup.ast.java.data.ResolutionStatus;
+import org.jboss.windup.ast.java.data.TypeReferenceLocation;
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.metadata.MetadataBuilder;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.config.phase.InitialAnalysisPhase;
+import org.jboss.windup.config.query.Query;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.reporting.service.ClassificationService;
+import org.jboss.windup.rules.apps.java.model.JavaClassModel;
+import org.jboss.windup.rules.apps.java.scan.ast.JavaTypeReferenceModel;
+import org.jboss.windup.rules.apps.java.scan.ast.TypeInterestFactory;
+import org.jboss.windup.rules.apps.java.service.JavaClassService;
+import org.jboss.windup.rules.apps.java.service.TypeReferenceService;
+import org.jboss.windup.rules.apps.javaee.model.JspSourceFileModel;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * Extracts type references from JSP files.
+ *
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class AnalyzeJSPFileRuleProvider extends AbstractRuleProvider
+{
+    public AnalyzeJSPFileRuleProvider()
+    {
+        super(MetadataBuilder.forProvider(AnalyzeJSPFileRuleProvider.class)
+                    .setPhase(InitialAnalysisPhase.class)
+                    .setHaltOnException(true));
+    }
+
+    // @formatter:off
+    @Override
+    public Configuration getConfiguration(GraphContext context)
+    {
+        return ConfigurationBuilder.begin()
+                .addRule()
+                .when(Query.fromType(JspSourceFileModel.class))
+                .perform(new ParseSourceOperation());
+    }
+    // @formatter:on
+
+    private class ParseSourceOperation extends AbstractIterationOperation<JspSourceFileModel>
+    {
+        @Override
+        public void perform(GraphRewrite event, EvaluationContext context, JspSourceFileModel sourceFile)
+        {
+
+            TypeReferenceService typeReferenceService = new TypeReferenceService(event.getGraphContext());
+
+            try
+            {
+                // Setup some basic details about the "Java Class"
+                // source root, is decompiled, javaclass. package name
+                sourceFile.setPackageName("");
+                sourceFile.setDecompiled(false);
+                JavaClassService javaClassService = new JavaClassService(event.getGraphContext());
+                JavaClassModel classModel = javaClassService.create();
+                classModel.setPackageName("");
+                classModel.setSimpleName(sourceFile.getFileName());
+                classModel.setQualifiedName(sourceFile.getFileName());
```

Comment:
Yes, as the package name is marked as empty. It's a bit of a hack as it is hard to know what the generated package name will really be.